src/hooks/use-authentication/index.ts

```diff
@@ -45,29 +55,23 @@ const useAuthentication = (
 		onUnauthenticated,
 	})
 
-	/**
-	 * Force sign in to hopefully resolve the error. The error is automatically
-	 * cleared in the process of initiating the login flow via `signIn`.
-	 *
-	 * Because `signOut` has to be invoked to fully log out of the provider, users
-	 * _should_ be re-signed in by this action without having to use the Cloud IDP
-	 * sign in screen.
-	 *
-	 * https://next-auth.js.org/tutorials/refresh-token-rotation#client-side
-	 */
-	useEffect(() => {
-		if (data?.error === AuthErrors.RefreshAccessTokenError) {
-			signOut()
-		}
-	}, [data?.error, signOut])
-
 	// Deriving booleans about auth state
 	const isLoading = status === 'loading'
 	const isAuthenticated = status === 'authenticated'
 	const showAuthenticatedUI = isAuthenticated
 	const showUnauthenticatedUI = !isLoading && !isAuthenticated
 	const preferencesLoaded = preferencesSavedAndLoaded()
 
+	/**
+	 * If the `showDebugUserMenuItems` feature flag is enabled, then `hasError`
+	 * is determined by whether or not the app's debugging state has an error in
+	 * it. Otherwise, we check whether for an error in the session `data` object.
+	 */
+	const enableDebugging = !!flags?.showDebugUserMenuItems
+	const hasError = enableDebugging
+		? debuggingState && debuggingState[DEBUGGING_ERROR_PROPERTY_NAME]
+		: data && data.error
```

Comment:
I think we should specifically scope this modal behavior to `AuthErrors.RefreshAccessTokenError` so we don't potentially swallow other auth errors.

Reply:
Is your concern about us not being alerted about them? If so, I think we should refine our monitoring and still show this modal for all auth errors since we don't have a good grasp of all the errors that have happened so far.

Reply:
Or if the error is of the token refresh type, we can render the modal with the 'your session has expired' copy...whereas if we got a different error and we wanted to render the modal, we could use your messaging that is just the 'encountered an auth error' message. 

I have a little bit of hesitation telling the user this is a routine session expiry when there could be something more / a bigger issue going on with the way we handle token revalidation 😬 I didn't feel that way until I saw Mike's error reproduction that seemed very different from our previous hypothesis of this issue being only related to multiple tabs going stale / revalidating.

Reply:
The error Mike was having falls in the same bucket as these. It looks like his `refresh_token` became out of date, which can happen multiple ways so it's hard to tell what exactly caused his error. I do feel confident that his error would have gone away if he had continued the sign in flow, but nothing told him to do that (which is where this modal comes in).

I think we should wait until we have more information before we change the messaging of the modal. The error that triggers all other errors has been the refresh token one, and we don't know the other ones yet. This modal will prevent the subsequent errors from happening at all, so our logs will be easier to follow and make an informed decision from.